[BE] 스터디방 조회

yejin013 opened the pull request:
스터디방 조회

- SortDefault를 사용해서 Sort.Direction.DESC 쓰고 싶었는데 적용이 안됨...; 무조건 url에 desc라고 쳐야지만 됨. 다들 이렇게 쓰라는데 이유를 알 수 없음;

yejin013 committed:
feat: 스터디방 조회

pageable를 활용한 스터디룸 조회

Related to #6

pjh4400 commented on server/src/main/java/com/oddok/server/domain/studyroom/api/StudyRoomController.java:
```diff
@@ -92,6 +94,18 @@ public String createBasic() {
+     * @return
+     */
+    @GetMapping
+    public ResponseEntity<Page<GetStudyRoomListEntityResponse>> get(@PageableDefault(size = 16) Pageable pageable,
```
@PageDefault(size=16, sort="id", direction = Sort.Direction.DESC) 이거 안됑??

yejin013 replied:
> @PageDefault(size=16, sort="id", direction = Sort.Direction.DESC) 이거 안됑??

응 그거 한 건데 안 먹히더라..? SortDefaults({sort="createAt", direction=Sort.Direction.DESC}, {sort="~~) 뭐 쨋든 이렇게 했는데

yejin013 committed:
Merge branch 'develop' of https://github.com/yejin013/Oddok into f/BE/get-study-room

yejin013 committed:
feat: category enum 생성

Related to #6

yejin013 committed:
feat: category로 조회

category를 눌렀을 때 조회

Related to #6

yejin013 committed:
comment: getStudyRooms 주석 추가

Related to #6

yejin013 committed:
feat: 스터디룸 검색

제목에 따른 검색

Related to #6

yejin013 committed:
refactor: 스터디룸 조회 if문 변경

Related to #6

pjh4400 commented on server/src/main/java/com/oddok/server/domain/studyroom/api/StudyRoomController.java:
```diff
@@ -92,6 +94,38 @@ public String createBasic() {
+    public ResponseEntity<Page<GetStudyRoomListEntityResponse>> get(@PageableDefault(size = 16) Pageable pageable,
+                                                                    @RequestParam(required = false) Boolean isPublic,
+                                                                    @RequestParam(required = false) String name) {
+        Page<GetStudyRoomListEntityResponse> studyRoomResponse = studyRoomService.getStudyRooms(pageable, isPublic, name).map(dtoMapper::toGetResponseList);
```
여기서 파라미터를 null 체크해서 isPublic 이 null 이면 true 로 지정해서 넘겨주면 서비스에서 null 체크를 하지 않아도 될것 같아! 그리고 name은 있으면 이름으로 검색하는 메소드를 서비스단에서 메소드를 따로 분리해서 호출해주면 어떨까?
getStudyRoomsByName 같은 느끰??

yejin013 replied:
> 여기서 파라미터를 null 체크해서 isPublic 이 null 이면 true 로 지정해서 넘겨주면 서비스에서 null 체크를 하지 않아도 될것 같아! 그리고 name은 있으면 이름으로 검색하는 메소드를 서비스단에서 메소드를 따로 분리해서 호출해주면 어떨까?
> getStudyRoomsByName 같은 느끰??

근데 true, false 한 번 더 체크해주긴 해야하는 게 false면 false의 값 뿐 아니라 true의 값도 가져와야 하기 때문에 해줄 수 밖에 없을 것 같아

pjh4400 replied:
이건 다 해결해주진 못하지만 체크 하나 줄일 수 있어서 말하자면 @RequestParams 에서 Boolean 형이 null 이면 자동으로 false 로 들어온대!

pjh4400 commented on server/src/main/java/com/oddok/server/domain/studyroom/api/StudyRoomController.java:
```diff
@@ -92,6 +94,38 @@ public String createBasic() {
+    public ResponseEntity<Page<GetStudyRoomListEntityResponse>> get(@PageableDefault(size = 16) Pageable pageable,
+                                                                    @RequestParam(required = false) String category,
+                                                                    @RequestParam(required = false) Boolean isPublic) {
+        Page<GetStudyRoomListEntityResponse> studyRoomResponse = studyRoomService.getStudyRooms(pageable, category, isPublic).map(dtoMapper::toGetResponseList);
```
아래랑 마찬가지로 여기서도 카테고리로 검색하면 호출하는 서비스를 getStudyRoomsByCategory 로 분리해주는 건 어떤지 한번 제안드려봅니당~

pjh4400 commented on server/src/main/java/com/oddok/server/domain/studyroom/entity/Category.java:
```diff
@@ -0,0 +1,14 @@
+package com.oddok.server.domain.studyroom.entity;
+
+public enum Category {
```
enum 으로 만든것 굿뜽👍

yejin013 committed:
refactor: get study-rooms 수정

isPublic 말고 다른 인자의 유무에 따라 함수 나누기

Related to #6

yejin013 committed:
fix: develop merge 충돌 해결

Merge branch 'develop' of https://github.com/yejin013/Oddok into f/BE/get-study-room
